### PR TITLE
Link-board

### PR DESCRIPTION
**Pull request by evwilkin**
comfort: 3
completeness: 3

Having trouble getting anything to write to the Posts db, it looks to be an issue of the Posts page not seeing the user_id but I've been unable to drill down to how to change this.

### CODE REVIEW

**Comment by evwilkin** [comment]
comfort: 4
completeness: 5

This homework was great to finally get my head around all the moving parts in the rails app, and get familiar with navigating between them all.  I have a better understanding of how the models, controllers, views, and routes work here and found myself knowing where to look for most problems.  I wish I had more time to complete the last couple challenges on this assignment to get that extra practice, but am glad that I was able to get lots of practice through the 5 listed below.
1. Added profile page at top right, only accessible when linked in, that lists user's posts and comments.  I'm having trouble understanding why on the Profile page the entire DB entry is printing out below all posts for each post and below all comments for each comment - we've seen this in rails console in the terminal but it doesn't show up on main Posts or Comments page which seem to loop through items in the same way?
2. Added count of total comments for each post on the index page.
3. Added modal in place of new comment page.
4. Added modal in place of new post page.
5. Added modal in place of login page.
6. Originally tried adding votes to the comments but I kept hitting an error - I think my routes need to be adjusted, but didn't have any luck when trying to split out the resources section at the top.  Left all applicable code in (some erb tags edited so it could be commented out correctly), any tips are appreciated!

File: app/views/comments/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          <%= f.text_area :content %>
+          <%= f.primary %>
+        <% end %>
+      </div>
+    </div><!-- /.modal-content -->
+  </div><!-- /.modal-dialog -->
+</div><!-- /.modal -->
+
+
+<!-- Show all comments for this post -->
```
Comment (jsakalys) [inline app/views/comments/index.html.erb:25]: To get the partial to show for each comment, make sure you nest it in the .each iterator for comments. Also, you must include the post that you are specifying in your locals object.

```
<!-- Show all comments for this post -->
<% @post.comment.each do |comment| %>
<div class="well">
  <p><%= comment.content %></p>
  <strong>By <%= comment.user.name %></strong>
 <%= render partial: 'comments/comment_votes', locals: {comment: comment, post: @post} %>
</div>
<% end %>
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Comment < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :post
+  # has_many :votes, as: :votable
```
Comment (jsakalys) [inline app/models/comment.rb:4]: Make sure you uncomment-out this here when re-implementing comment voting

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  # You can have the root of your site routed with "root"
+  root 'posts#index'
+
+  resources :posts, only: [:new, :create, :show] do
+    resources :comments, only: [:index, :new, :create]
+    post 'upvote' => 'posts#upvote'
+    post 'downvote' => 'posts#downvote'
+  end
+
+  # resources :comments, only: [:index, :new, :create] do
```
Comment (jsakalys) [inline config/routes.rb:15]: Also bring this back in now!

**Comment by jsakalys** [comment]
Hi Evan, first of all, fantastic work here! This is very cohesive and polished. To implement comment voting, please try my suggestions above. If you still can't get it to work, please come see me. You will have to keep in mind that you'll need to edit your controller slightly to accommodate both posts and comments when voting. I think you can take it from here though. Best of luck!

**Pull request closed without merging**